ruoyi-common/src/test/java/com/ruoyi/common/support/ConvertTest.java

```diff
@@ -0,0 +1,202 @@
+package com.ruoyi.common.support;
+
+import java.nio.ByteBuffer;
+
+import org.junit.Test;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.nio.charset.Charset;
+import java.util.Set;
+import java.util.TreeSet;
+
+import static com.ruoyi.common.support.Convert.*;
+import static org.junit.Assert.*;
+
+public class ConvertTest {
+
+    @Test
+    public void testToStr() {
+        assertEquals("bar", toStr(null, "bar"));
+        assertEquals("123", toStr(123, "bar"));
+        assertEquals("foo", toStr("foo"));
```

Comment:
Out of curiosity more than anything, i'd quite like to see an Assert here for when we call the toStr method but pass in two nulls. 

The method is overloaded to revert to assigning null as the defaultValue if one isn't passed, but for redundancy it might be nice to show that null is still returned if a null value is passed for the defaultValue.

Reply:
If we call toStr(null, null) it will return null.
This is the same (in terms of testing lines) as calling toStr(null, "bar") and returning "bar" or have I missed the point?

Reply:
Indeed, the line has been covered in terms of what is actually being executed. It may be a question of testing style, but I quite like the thoroughness of reaching the null return directly, as well as indirectly with the overloaded method.

Reply:
Are you saying there is a point in returning null instead of "bar"? I could see that might be the case if passing null, as opposed to a string, would throw an exception for example.
There is no indirect returning of null in the proposed tests ie. there is no call to toStr(null).

Reply:
I would say so, yes. We get the same return from toStr(null) as we do from toStr(null, null), but the former will get this return (or exception) by calling a separate function. 
Therefore I would probably argue that reaching this same end state by both variants of the toStr() function would result in a more robust test method.

Reply:
So.... what am i doing for this? I got all the lines covered for the method. Do you want me to still test for the (null,null) values?

Reply:
I am happy with the proposed tests which cover all the lines. I'd only be in favour of adding an extra test with a null input if it exercised the code differently.

Reply:
@paulkennethkent what should I do about this?